sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/CustomFormModel.java

```diff
@@ -70,9 +72,10 @@ public CustomFormModel(final String modelId, final CustomFormModelStatus modelSt
         this.modelStatus = modelStatus;
         this.requestedOn = requestedOn;
         this.completedOn = completedOn;
-        this.submodels = submodels;
-        this.modelError = modelError;
-        this.trainingDocuments = trainingDocuments;
+        this.submodels = submodels == null ? null : Collections.unmodifiableList(new ArrayList<>(submodels));
+        this.modelError = modelError == null ? null : Collections.unmodifiableList(new ArrayList<>(modelError));
+        this.trainingDocuments = trainingDocuments == null ? null
+            : Collections.unmodifiableList(new ArrayList<>(trainingDocuments));
```

Comment:
Same here - don't have to create a new ArrayList. Do the same in all other places too.